nav2_smac_planner/src/analytic_expansion.cpp

```diff
@@ -60,101 +60,121 @@ typename AnalyticExpansion<NodeT>::NodePtr AnalyticExpansion<NodeT>::tryAnalytic
     const Coordinates node_coords =
       NodeT::getCoords(
       current_node->getIndex(), _collision_checker->getCostmap()->getSizeInCellsX(), _dim_3_size);
-    closest_distance = std::min(
-      closest_distance,
-      static_cast<int>(NodeT::getHeuristicCost(node_coords, goal_node->pose)));
-
-    // We want to expand at a rate of d/expansion_ratio,
-    // but check to see if we are so close that we would be expanding every iteration
-    // If so, limit it to the expansion ratio (rounded up)
-    int desired_iterations = std::max(
-      static_cast<int>(closest_distance / _search_info.analytic_expansion_ratio),
-      static_cast<int>(std::ceil(_search_info.analytic_expansion_ratio)));
-
-    // If we are closer now, we should update the target number of iterations to go
-    analytic_iterations =
```

Comment:
This entire block (https://github.com/ros-planning/navigation2/blob/main/nav2_smac_planner/src/analytic_expansion.cpp#L60-L76) is a preamble, we shouldn't be messing with that. We don't want to run analytic expansions each iteration, so this is what throttles it within reason. The same logic applies here. We only run the full suite of different angled analytic expansions when our desired iterations comes up. Then, we trigger all of them. 

So, this should be up front, before the loop. We only loop `if (analytic_iterations <= 0)`, else we do nothing. Do you agree?

Reply:
I agree, in addition to that the distance heuristic is precomputed, it does not make any sense either to have it in the loop since the value will always be same.